torch/distributed/checkpoint/_nested_dict.py

```diff
@@ -40,7 +47,13 @@ def flat_copy(path: OBJ_PATH, value: STATE_DICT_ITEM) -> None:
         flattened[new_fqn] = value
         mappings[new_fqn] = path
 
-    traverse_state_dict(state_dict, flat_copy)
+    use_v_2_3 = (
```

Comment:
Shall we add a doc comment on the discrepancy we are solving. I see that you try to hit that by v_2_3, but a new devloper might not be able to understand the context.